README.md

```diff
@@ -15,25 +16,21 @@ Checks definitions: see the [walrus].
 [walrus]: /path/to/walrus.jpg
 ```
 
-Uses [link-check](https://github.com/tcort/link-check) to check URLs against the real internet.
-
-Options: `Object` or `string`. Optional.
+Uses [check-links](https://github.com/transitive-bullshit/check-links) to check URLs for liveness (requires an internet connection).
 
-An options `Object` can have the following properties:
+Options: `Object`. Optional. May contain any of the following properties:
 
+- **concurrency** `number` - Maximum number of urls to resolve concurrently (optional, default `8`)
 - **baseUrl** `string` - Used as the base URL against which relative URLs are checked.
   For example, with `baseUrl: 'https://www.github.com'`, the relative URL `/davidtheclark` is checked as `https://www.github.com/davidtheclark`.
   **By default, relative URLs are ignored: you must provide this option to check them.**
-- **cache** `Object` - By default, URLs are cached internally to avoid repeated checks.
-  If you want to manage the cache yourself (maybe even write it to disk between runs), you can provide a cache `Object` that will be read from and written to.
-  The cache will be populated with properties whose keys are URLs are values are `'alive'` or `'dead'`.
 - **skipOffline** `boolean` - Default: `false`.
   By default, if you are offline when you run the check you will receive a warning.
   If you want to let offline runs quietly pass, set this option to `true`.
 
-An options `string` is interpreted as `baseUrl` (above).
+Any other properties will be passed to as options to [got](https://github.com/sindresorhus/got#options), such as customizing retry logic, setting a `baseUrl`, specifying a custom `cache`, custom headers, etc.
```

Comment:
I think I'd prefer to create a `gotOptions` property that you can use to explicitly pass options got.

I think the recommendation to use got's cache makes sense; but I'm also seeing the use of p-memoize in check-links: https://github.com/transitive-bullshit/check-links/blob/cf5fbcf0fc0bd150097034887b7c132384794549/index.js#L53-L55. Won't p-memoize end up overriding whatever your got cache configuration?

Reply:
The `p-memoize` vs got `cache` is a very good point. My gut says that 95% of the time we'd want the functionality of `p-memoize` vs relying on `cache` at the individual HTTP layer since under the hood we're invoking both HEAD and GET requests and having that `cache` duplication is a bit weird. Generally when I come across this type of interface ambiguity for potential ways of handling a problem, I ask myself what's the easiest way of handling the 95%  expected use case and I think `p-memoize` seems to answer that for me since I don't want to assume users will intelligently pass an LRU-ish `cache` along.

I'll give some more thought to this and follow-up with a proposed compromise tomorrow :) I just wanted to get my initial thoughts out tonight.

I'm happy to make the `gotOptions` change explicit instead of implicit as things currently stand.

Reply:
I agree that p-memoize is also typically sufficient, but maybe there should be an option to turn it off or provide your own object or Map.

> I'm happy to make the gotOptions change explicit instead of implicit as things currently stand.

👍